src/core/utils/helpers/modal-helpers.ts

```diff
@@ -0,0 +1,54 @@
+import { getModalIds } from "./general";
+
+export const containsDueDateModalString = (modalString: string) => {
+  const { dueDateModal } = getModalIds();
+  // regex for finding duedatemodal concatenated with date string from modal query param
+  const regex = new RegExp(`${dueDateModal}\\d{4}-\\d{2}-\\d{2}`, "g");
+
+  const dateFound = modalString.match(regex);
+  if (!dateFound) {
+    return null;
+  }
+  return dateFound[0];
+};
+
+export const dateFromDueDateModalString = (modalString: string) => {
+  // regex for finding duedatemodal concatenated with date string from modal query param
+  const regex = /\d{4}-\d{2}-\d{2}/g;
+
+  const dateFound = modalString.match(regex);
+
+  if (!dateFound) {
+    return null;
+  }
+  return dateFound[0];
+};
+export const idFromLoanDetailsModalString = (modalString: string) => {
+  // regex for finding duedatemodal concatenated with date string from modal query param
+  const regexIdentifier = /\d{13}/g;
+  const regexFaust = /\d{8}/g;
+
+  let idFound = modalString.match(regexIdentifier);
+  if (!idFound) {
+    idFound = modalString.match(regexFaust);
+    if (!idFound) {
+      return null;
+    }
+  }
+  return idFound[0];
+};
```

Comment:
I am not completely shure what you are trying to match. 
If `modalString` is more than 13 digits it will also match that (if it is possible to be more than 13).
Can there be characters before and/or after?
I think it is more clear if you have one regex. without knowing the problem scope exactly you could also write:
```
const result = modalString.match(/(^\d{8}$)|(^\d{13}$)/);

Reply:
there can be characters before and after, does that change this question?